### ISSUE TITLE
Unable to run with `.cjs` files as `.js`

### ISSUE THREAD

**Issue by JounQin** [stylelint/stylelint#5432]
> Clearly describe the bug

As title

> Which rule, if any, is the bug related to?

e.g. `number-leading-zero`

> What code is needed to reproduce the bug?

```js
// test.cjs
module.exports = 1
```

> What stylelint configuration is needed to reproduce the bug?

No need

> Which version of stylelint are you using?

`13.13.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint test.cjs`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

No error

> What actually happened (e.g. what warnings or errors did you get)?

```log
test.cjs
 1:1  ✖  Unknown word   CssSyntaxError
```

---

If I rename `test.cjs` to `test.js`, there will be no error.

**Comment by jeddy3** [comment]
@JounQin Thanks for the report and for using the template.

This is likely an issue upstream in `postcss-syntax` (what we use to infer syntax). In [`14.0.0`](https://github.com/stylelint/stylelint/issues/5205) we intend to move to explicit configuration, so this should no longer be an issue.

In the meantime, you can use the [`--syntax` option](https://stylelint.io/user-guide/usage/options#syntax) (which will become `custom-syntax` in `14.0.0` to explicit set the syntax for `.cjs` files:

```
stylelint "**/*.cjs" --syntax css-in-js
```

**Comment by JounQin** [comment]
I meet this issue because of [vscode-stylelint](https://github.com/stylelint/vscode-stylelint), so can the workaround be used into `vscode-stylelint`?

**Comment by jeddy3** [comment]
I'm not sure. Maybe [this option](https://github.com/stylelint/vscode-stylelint#stylelintsyntax).

Alternatively, you can limit what the extension validates using [this option](https://github.com/stylelint/vscode-stylelint#stylelintvalidate), e.g. :

```
{
  stylelint.validate: ["css"]
}
```

Or ignore `.cjs` files using the [`ignoreFiles`](https://stylelint.io/user-guide/configure#ignorefiles) option in stylelint, which the extension will respect.

**Comment by JounQin** [comment]
OK, then I have to use `.stylelintignore` for now, the other two options seem not good to me.
